test_common/harness/errorHelpers.cpp

```diff
@@ -594,33 +627,7 @@ cl_int OutputBuildLogs(cl_program program, cl_uint num_devices,
         unsigned int i;
         for (i = 0; i < num_devices; i++)
         {
-
-            // Get the build status
-            cl_build_status build_status;
-            error = clGetProgramBuildInfo(
-                program, device_list[i], CL_PROGRAM_BUILD_STATUS,
-                sizeof(build_status), &build_status, &size_ret);
-            test_error(error, "Unable to query build status");
-
-            // If the build failed then log the status, and allocate the build
-            // log, log it and free it
-            if (build_status != CL_BUILD_SUCCESS)
-            {
-
-                log_error("ERROR: CL_PROGRAM_BUILD_STATUS=%d\n",
-                          (int)build_status);
-                error = clGetProgramBuildInfo(program, device_list[i],
-                                              CL_PROGRAM_BUILD_LOG, 0, NULL,
-                                              &size_ret);
-                test_error(error, "Unable to query build log size");
-                char *build_log = (char *)malloc(size_ret);
-                error = clGetProgramBuildInfo(program, device_list[i],
-                                              CL_PROGRAM_BUILD_LOG, size_ret,
-                                              build_log, &size_ret);
-                test_error(error, "Unable to query build log");
-                log_error("ERROR: CL_PROGRAM_BUILD_LOG:\n%s\n", build_log);
-                free(build_log);
-            }
+            OutputBuildLog(program, device_list[i]);
```

Comment:
If `OutputBuildLog` generates an error do we want to propagate it back to the caller, or are we OK "swallowing" the error and returning `CL_SUCCESS`?

Perhaps this isn't terribly critical because `OutputBuildLog` only generates an error if it is unable to query the build log, and does NOT return an error if the build failed, so generating an error will be a very exceptional condition, but as written this will be a small behavioral difference compared to the previous code.

Reply:
Good point; it seems other parts of the code propagate build log query errors upwards, so I've updated the patch to do so here and in `get_program_with_il`.